diff --git a/src/main/java/com/cryptocurrency/cryptoAPI/service/UserService.java b/src/main/java/com/cryptocurrency/cryptoAPI/service/UserService.java
index 74b3af1..dd05cbf 100644
--- a/src/main/java/com/cryptocurrency/cryptoAPI/service/UserService.java
+++ b/src/main/java/com/cryptocurrency/cryptoAPI/service/UserService.java
@@ -40,11 +40,11 @@ public class UserService extends CrudService<User> implements UserDetailsService
 
         Currency currency = new Currency(symbol, currencyService.calcValueFor(symbol).get(symbol), user);
         //NUM07-J. Do not attempt comparisons with NaN
-        if(currency.getValue() == 0)
+        if(currency == null)
             return false;
 
         //NUM09-J. Do not use floating-point variables as loop counters
-        for(float i=amount; i>0f; i--)
+        for(int i=amount; i>0; i--)
         addCurrency(username, currency);
         user.setMoney(currentFunds - intendedBuy);
         repository.save(user);
diff --git a/src/main/java/com/cryptocurrency/cryptoAPI/util/ApiConstants.java b/src/main/java/com/cryptocurrency/cryptoAPI/util/ApiConstants.java
index 26e0210..d91baab 100644
--- a/src/main/java/com/cryptocurrency/cryptoAPI/util/ApiConstants.java
+++ b/src/main/java/com/cryptocurrency/cryptoAPI/util/ApiConstants.java
@@ -2,7 +2,7 @@ package com.cryptocurrency.cryptoAPI.util;
 
 //OBJ10-J. Do not use public static nonfinal fields
 public class ApiConstants {
-    public static String CURRENCY = "PLN";
-    public static String API_URL = "https://min-api.cryptocompare.com/";
-    public static String API_KEY = "f693bb0c9237e079c6dc162f34f9870e65ee6ce9dc469e45c4f443c9d9816dec";
+    public static final String CURRENCY = "PLN";
+    public static final String API_URL = "https://min-api.cryptocompare.com/";
+    public static final String API_KEY = "f693bb0c9237e079c6dc162f34f9870e65ee6ce9dc469e45c4f443c9d9816dec";
 }
